src/vs/workbench/contrib/chat/browser/chatContentParts/chatReferencesContentPart.ts

```diff
@@ -392,8 +403,11 @@ class CollapsibleListRenderer implements IListRenderer<IChatCollapsibleListItem,
 		}
 
 		if (this.menuId) {
+			if (data.state !== undefined) {
+				chatEditingWidgetFileStateContextKey.bindTo(templateData.contextKeyService).set(data.state);
+			}
 			const actionBarContainer = $('.chat-collapsible-list-action-bar');
-			templateData.toolbar = templateData.templateDisposables.add(this.instantiationService.createInstance(MenuWorkbenchToolBar, actionBarContainer, this.menuId, { menuOptions: { shouldForwardArgs: true, arg: arg } }));
+			templateData.toolbar = templateData.templateDisposables.add(templateData.instantiationService.createInstance(MenuWorkbenchToolBar, actionBarContainer, this.menuId, { menuOptions: { shouldForwardArgs: true, arg: arg } }));
```

Comment:
You can create the toolbar on the template, so it only has to be created once, then will update when you update the context keys

Reply:
Also it looks like the lifecycle is off, adding to `templateDisposables` inside `renderElement`